src/components/modal/index.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect } from 'react';
+import classnames from 'classnames';
+import CloseIcon from '../icons/close';
+import IconButton from '../../components/icon-button';
+import './modal.css';
+import utils from '@quantumblack/kedro-ui/lib/utils';
+
+/**
+ * Generic Kedro Modal
+ */
+const Modal = ({ title, onClose, visible, message, children }) => {
+  const handleKeyDown = (event) => {
+    utils.handleKeyEvent(event.keyCode, {
+      escape: onClose,
+    });
```

Comment:
Appreciate you using the utils here from `kedro-ui` though perhaps we want to move away from that since we're also moving away from that library as a whole? What do you think?

You could get the same behavior using this:

```suggestion
if (event.keyCode === 27) {
  onClose();
}
```

Reply:
Sure, i will do this. There is another ticket to handle utils but since it's being used here I will address it here itself :) thanks!